src/nwbinspector/checks/nwbfile_metadata.py

```diff
@@ -197,10 +197,12 @@ def check_subject_id_exists(subject: Subject):
 
 @register_check(importance=Importance.BEST_PRACTICE_SUGGESTION, neurodata_type=Subject)
 def check_subject_sex(subject: Subject):
-    """Check if the subject sex has been specified, if one exists."""
+    """Check that the subject sex has been correctly specified."""
     if subject and not subject.sex:
         return InspectorMessage(message="Subject.sex is missing.")
-    elif subject.sex not in ("M", "F", "O", "U"):
+    if subject.species in ("Caenorhabditis elegans", "C. elegans") and subject.sex not in ("XO", "XX"):
+        return InspectorMessage(message="For C. elegans, Subject.sex should be 'XO' or 'XX'.")
+    if subject.sex not in ("M", "F", "O", "U"):
```

Comment:
```suggestion
    elegans_references = ("Caenorhabditis elegans", "C. elegans")
    if subject.species in elegans_references  and subject.sex not in ("XO", "XX"):
        return InspectorMessage(message="For C. elegans, Subject.sex should be 'XO' or 'XX'.")
    if subject.species not in elegans_references and subject.sex not in ("M", "F", "O", "U"):
```

Flatter is better than nested, according to Python zen (i.e., reduce cyclomatic complexity)

If we ever need more special cases for other species in the far future, we can start using switch statements in Python 3.10

Reply:
Even what I suggested here is more logic than necessary; added new approach with helper functions that will scale well with adding more special cases to future species, if necessary.